### DISCUSSION TITLE
Hide instrumentationKey from the client

### DISCUSSION THREAD

**Discussion by HiZackDavid** [IvanJosipovic/BlazorApplicationInsights#231] [Q&A] [answered]
Ivan provided a beautiful explanation on how to set up "Application Insights for Blazor web applications".
That being said, in the example provided, the instrumentation Key is visible to users when we inspect it in the browser.
![image](https://github.com/IvanJosipovic/BlazorApplicationInsights/assets/87014651/1e997cb8-a76a-4e6b-af39-a0af46d1f6b4)

I was wondering if there was a way to set up Applications Insights but hide the Instrumentation Key to users in the browser ?

I will leave here the current set up I have for Application Insights in Blazor : 
**appsettings.json** ("..." means that I hid the content)
```json
{
  ...
  "Telemetry": {
    ...
    "InstrumentationKey": "..."
  }
  ...
}
```

**wwwroot\index.html**
```html
<!DOCTYPE html>
<html lang="en">
    <head>
        ...
        <script>
             ...
             cfg: {
                    instrumentationKey: "", // leave this blank
                    disableFetchTracking: false,
                    enableCorsCorrelation: true,
                    enableRequestHeaderTracking: true,
                    enableResponseHeaderTracking: true
             }
             ...
        </script>
    </head>
    <body>
    </body>
</html>
```

**Program.cs**
```csharp
using BlazorApplicationInsights;
using BlazorApplicationInsights.Sample;
using Microsoft.AspNetCore.Components.Web;
using Microsoft.AspNetCore.Components.WebAssembly.Hosting;

public class Program
{
    public static async Task Main(string[] args)
    {
        ...
        builder.Services.AddBlazorApplicationInsights(async applicationInsights =>
        {
            var instrumentationKey = builder.Configuration.GetSection("Telemetry").GetValue<string>("InstrumentationKey");
            await applicationInsights.SetInstrumentationKey(
                instrumentationKey ?? "..."
            );
        });
        ...
    }
}
```

Note : 
- BlazorApplicationInsights was already installed
- @using BlazorApplicationInsights; was already added to _Imports.razor

**Comment by IvanJosipovic** [comment, marked as answer]
There is no way to hide the instrumentation key from the client, it is exposed in a number of places like JavaScript and every http request to AppInsights.
